projects/ui/src/lib/components/po-popup/po-popup.component.ts

```diff
@@ -159,7 +159,12 @@ export class PoPopupComponent extends PoPopupBaseComponent {
   }
 
   private onScroll = ({ target }): void => {
-    if (this.showPopup && target.className !== 'po-popup-container' && this.actions.length < 6) {
+    if (
+      this.showPopup &&
+      this.popupRef &&
+      this.popupRef.nativeElement &&
```

Comment:
Para melhorar a legibilidade e manutenção futura do código, você pode simplificar e tornar mais explícitas as condições do 'if' 

Segue sugestão de melhoria 


// Aqui você utiliza a desestruturação para extrair 'popupRef' e 'showPopup' do this;
const { showPopup, popupRef } = this;


// aqui você utiliza o optional chaining para reduziar o if, se não existir popupRef o if já é cancelado com sucesso; 

if (showPopup && popupRef?.nativeElement && !popupRef.nativeElement.contains(target)) {
    this.close();
}

Reply:
Ajustado como solicitado.